internal/testutils/tempdir.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// TempDir is a temporary directory which is set up with a direcotry structure with some files for testing.
+type TempDir struct {
```

Comment:
Use TempDir in https://github.com/ianlewis/todos/blob/main/internal/cmd/todos/app_test.go